### PR TITLE
make both category subroutes resolve category objects in the same way

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/3941 by making sure both '/blog/category/.../` and the JSON fetch request deeper in that route resolve category slugs to category objects in the exact same way.

### CODE REVIEW

File: network-api/networkapi/wagtailpages/models.py [hunk 1]
```diff
@@ -735,8 +735,21 @@ def entries_by_tag(self, request, tag, *args, **kwargs):
     category routes
     """
 
+    # helper function to resolve category slugs to actual objects
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/models.py]: before, this code only existed in the `category/somecategory` route handler, so it never kicked in for the "load more" request (which uses a different route) and so the actual result fetching completely failed.

**Comment by kristinashu** [comment]
Yes, made a test page here https://foundation-mofostaging-pr-3944.herokuapp.com/en/blog/category/advocacy/ and it seems to work!

**Comment by mmmavis** [comment]
@Pomax redirecting bad category url doesn't seem to work. https://foundation-mofostaging-pr-3944.herokuapp.com/en/blog/category/foo/ gives me `500`

**Comment by Pomax** [comment]
Good catch @mmmavis, fixed!

**Pull request merged**